source/extensions/filters/http/fault/fault_filter.cc

```diff
@@ -71,6 +71,20 @@ FaultFilter::~FaultFilter() { ASSERT(!delay_timer_); }
 // if we inject a delay, then we will inject the abort in the delay timer
 // callback.
 Http::FilterHeadersStatus FaultFilter::decodeHeaders(Http::HeaderMap& headers, bool) {
+  // Route-level configuration overrides filter-level configuration
+  // NOTE: We should not use runtime when reading from route-level
+  // faults. In other words, runtime is supported only when faults are
+  // configured at the filter level.
+  if (callbacks_->route() && callbacks_->route()->routeEntry()) {
+    const std::string name = Extensions::HttpFilters::HttpFilterNames::get().FAULT;
+    const auto* route_entry = callbacks_->route()->routeEntry();
+
+    const FaultSettings* per_route_settings =
+        route_entry->perFilterConfigTyped<FaultSettings>(name)
+            ?: route_entry->virtualHost().perFilterConfigTyped<FaultSettings>(name);
+    config_->updateFaultSettings(per_route_settings);
```

Comment:
Aren't you effectively updating a global shared pointer here? I don't think this is right? Can you make sure `FaultFilterConfigSharedPtr` is actually `FaultFilterConfigConstSharedPtr` to catch this?

Reply:
ack.. good catch..

Reply:
now that I think about it, the reason I did this was because I wanted to avoid allocation in the request path. I couldn't create an identical config object for RouteSpecific.. because I don't have access to runtime/stats/scope etc in that API Call.

Reply:
Hmm. Then this is the same problem as Lua. We need to fix this now I guess and be able to pass the full FactoryContext into the factory function. Can you take a look at that?